--- /home/kr2/.nix-defexpr/channels/nixos/nixos/modules/installer/cd-dvd/installation-cd-base.nix	1970-01-01 01:00:01.000000000 +0100
+++ installation-cd-rmk35.nix	2014-12-20 23:47:04.464105585 +0000
@@ -7,13 +7,17 @@
 
 {
   imports =
-    [ ./channel.nix
-      ./iso-image.nix
+    [ /home/kr2/.nix-defexpr/channels/nixos/nixos/modules/installer/cd-dvd/channel.nix
+      ./iso-image-rmk35.nix
 
-      # Profiles of this basic installation CD.
-      ../../profiles/all-hardware.nix
-      ../../profiles/base.nix
-      ../../profiles/installation-device.nix
+      # Profiles of the basic installation CD.
+      # Replaced relative paths with absolute, as I have the patched file elsewhere
+      /home/kr2/.nix-defexpr/channels/nixos/nixos/modules/profiles/all-hardware.nix
+      /home/kr2/.nix-defexpr/channels/nixos/nixos/modules/profiles/base.nix
+      /home/kr2/.nix-defexpr/channels/nixos/nixos/modules/profiles/installation-device.nix
+
+      # System configuration
+      /etc/nixos/live-configuration.nix
     ];
 
   # ISO naming.
@@ -39,8 +43,9 @@
   # Add Memtest86+ to the CD.
   boot.loader.grub.memtest86.enable = true;
 
-  # Get a console as soon as the initrd loads fbcon on EFI boot.
-  boot.initrd.kernelModules = [ "fbcon" ];
+  # Get a console as soon as the initrd loads fbcon on EFI boot
+  # I have my files on a UDF partition, hence need UDF
+  boot.initrd.kernelModules = [ "fbcon" "udf" ];
 
   # Add support for cow filesystems and their utilities
   boot.supportedFilesystems = [ "zfs" "btrfs" ];
